Content.Shared/Ninja/Systems/NinjaGlovesSystem.cs

```diff
@@ -0,0 +1,325 @@
+    }
+
+    // can't predict PNBC existing so only done on server.
+    protected virtual void OnDrain(EntityUid uid, NinjaDrainComponent comp, InteractionAttemptEvent args) { }
+
+    private void OnDrainDoAfter(EntityUid uid, NinjaDrainComponent comp, DoAfterEvent args)
+    {
+        if (args.Cancelled || args.Handled || args.Args.Target == null)
+            return;
+
+        var user = args.Args.User;
+        var target = args.Args.Target.Value;
+
+        _ninja.TryDrainPower(user, comp, target);
+    }
+
+    private void OnDownload(EntityUid uid, NinjaDownloadComponent comp, InteractionAttemptEvent args)
+    {
+        if (!GloveCheck(uid, args, out var gloves, out var user, out var target))
+            return;
+
+        if (gloves.Busy || !TryComp<TechnologyDatabaseComponent>(target, out var database))
+            return;
+
+        // fail fast if theres no tech right now
+        if (database.TechnologyIds.Count == 0)
+        {
+            ClientPopup(Loc.GetString("ninja-download-fail"), user);
+            return;
+        }
+
+        var doafterArgs = new DoAfterEventArgs(user, comp.DownloadTime, target: target, used: uid)
+        {
+            BreakOnDamage = true,
+            BreakOnStun = true,
+            BreakOnUserMove = true,
+            MovementThreshold = 0.5f
+        };
+
+        _doafter.DoAfter(doafterArgs);
+        gloves.Busy = true;
+    }
+
+    private void OnDownloadDoAfter(EntityUid uid, NinjaDownloadComponent comp, DoAfterEvent args)
+    {
+        if (args.Cancelled || args.Handled)
+            return;
+
+        var user = args.Args.User;
+        var target = args.Args.Target;
+
+        if (!TryComp<NinjaComponent>(user, out var ninja)
+            || !TryComp<TechnologyDatabaseComponent>(target, out var database))
+            return;
+
+        var gained = _ninja.Download(ninja, database.TechnologyIds);
+        var str = gained == 0
+            ? Loc.GetString("ninja-download-fail")
+            : Loc.GetString("ninja-download-success", ("count", gained), ("server", target));
+
+        _popups.PopupEntity(str, user, user, PopupType.Medium);
+    }
+
+    private void OnTerror(EntityUid uid, NinjaTerrorComponent comp, InteractionAttemptEvent args)
+    {
+        if (!GloveCheck(uid, args, out var gloves, out var user, out var target)
+            || !TryComp<NinjaComponent>(user, out var ninja))
+            return;
+
+        if (gloves.Busy || !IsCommsConsole(target))
+            return;
+
+        // can only do it once
+        if (ninja.CalledInThreat)
+        {
+            _popups.PopupEntity(Loc.GetString("ninja-terror-already-called"), user, user);
+            return;
+        }
+
+        var doafterArgs = new DoAfterEventArgs(user, comp.TerrorTime, target: target, used: uid)
+        {
+            BreakOnDamage = true,
+            BreakOnStun = true,
+            BreakOnUserMove = true,
+            MovementThreshold = 0.5f
+        };
```

Comment:
```suggestion
        {
            RaiseOnUser = false,
            RaiseOnTarget = false,
            BreakOnDamage = true,
            BreakOnStun = true,
            BreakOnUserMove = true,
            MovementThreshold = 0.5f
        };
```

Reply:
See if you can block the APC ui from opening/closing on interact while the gloves are on as well. It can become a hassle if you're trying to drain from the APC multiple times and the UI keeps popping up in your face.